troubleshooting/tools/tracevol.py

```diff
@@ -562,6 +563,88 @@ def get_fsname_from_pvdata(arg, pvdata):
         sys.exit()
     return fsname
 
+def kube_client(*commands):
+    """
+    Executes a kubectl/oc command with the provided arguments and returns the JSON output.
+    Note: Do not pass `-o json` as an argument, this helper adds it automatically.
+
+    Example:
+        kube_client("get", "pvcs")
+
+    Args:
+        *commands: Additional command-line arguments to pass to the Kubernetes or OpenShift command.
+    """
+    arg = ARGS
+    cmd = [arg.command]
+    if arg.kubeconfig != "":
+        cmd += ["--kubeconfig", arg.kubeconfig]
+    cmd += ["--namespace", arg.namespace]
+    cmd += filter(lambda x: x not in ('-o', 'json'), commands)
+    cmd += ["-o", "json"]
+
+    with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as out:
+        stdout, stderr = out.communicate()
+
+    if stderr is not None:
+        if arg.debug:
+            print(f"failed to execute command {commands} due to {stderr}")
+        sys.exit()
+    try:
+        data = json.loads(stdout)
+    except ValueError as err:
+        print(err, stdout)
+        sys.exit()
+
+    return data
+
+def parse_volume_snapshots(volsnapshots):
+    """
+    Parses a list of volume snapshots and prints a table with their details.
+    """
+    cephfs_snap_tbl = prettytable.PrettyTable()
+    cephfs_snap_tbl.title = "CephFS Volume Snapshots"
+    cephfs_snap_tbl.field_names = ["Name", "PVC", "Subvolume Name", "Snapshot ID"]
+
+    rbd_snap_tbl = prettytable.PrettyTable()
+    rbd_snap_tbl.title = "RBD Volume Snapshots"
+    rbd_snap_tbl.field_names = ["Name", "PVC", "Image Name", "Snapshot ID"]
+
+    for snapshot in volsnapshots['items']:
+        snap_name = snapshot['metadata']['name']
+        pvc_name = snapshot['spec']['source']['persistentVolumeClaimName']
+
+        # Fetch the PVC and its PV, useful later
+        pvc = kube_client("get", "pvc", pvc_name)
+        pv_data = kube_client("get", "pv", pvc['spec']['volumeName'])
+
+        # Find the type of the PVC (RBD or CephFS)
+        fs_type = "CephFS" if "fsName" in pv_data['spec']['csi']['volumeAttributes'] else "RBD"
```

Comment:
what happens if there are non-csi pvc or pvc's belongs to other storage provider?

Reply:
Updated it like: If the encountered PVC is not a CSI volume, we skip it. 

> belongs to other storage provider?

The existing checks in the script only contain cases for either CephFS or RBD. I can refactor the detection logic to be more precise in the upcoming PR?